poh/src/poh_recorder.rs

```diff
@@ -279,11 +280,16 @@ impl PohRecorder {
                 last_reported_slot_for_pending_fork: Arc::default(),
                 is_exited,
                 entries: Vec::with_capacity(64),
+                track_transaction_indexes: false,
             },
             working_bank_receiver,
         )
     }
 
+    pub fn track_transaction_indexes(&mut self) {
+        self.track_transaction_indexes = true;
+    }
```

Comment:
Apologies for my very late review / I'm glad you didn't wait for me. Overall, this seems like good cleanup and makes sense. One minor item:

> track_transaction_indexes doesn't change from bank-to-bank, it's static after start up.

Given this, I'm thinking `track_transaction_indexes` should instead be an input to the constructor instead of a `&mut self` modifier. I'm happy to do the follow-on cleanup, but figured I'd double check here to make sure I'm not missing anything